aws/resource_aws_sagemaker_endpoint_test.go

```diff
@@ -0,0 +1,470 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerEndpointConfigTags(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerEndpointExists("aws_sagemaker_endpoint.foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_endpoint.foo", "tags.%", "1"),
+					resource.TestCheckResourceAttr("aws_sagemaker_endpoint.foo", "tags.foo", "bar"),
+				),
+				ExpectError: regexp.MustCompile(`.*unexpected state 'Failed', wanted target 'InService'.*`),
+			},
+			{
+				Config: testAccSagemakerEndpointConfigTagsUpdate(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerEndpointExists("aws_sagemaker_endpoint.foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_endpoint.foo", "tags.%", "1"),
+					resource.TestCheckResourceAttr("aws_sagemaker_endpoint.foo", "tags.bar", "baz"),
+				),
+			},
+			{
+				ResourceName:      "aws_sagemaker_endpoint.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerEndpoint_update(t *testing.T) {
+	// Cannot update failed endpoint
+	t.Skip()
+
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerEndpointConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerEndpointExists("aws_sagemaker_endpoint.foo"),
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_endpoint.foo",
+						"name",
+						fmt.Sprintf("terraform-testacc-sagemaker-endpoint-%s", rName)),
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_endpoint.foo",
+						"endpoint_config_name",
+						"terraform-testacc-sagemaker-endpoint-config-foo"),
+				),
+				ExpectError: regexp.MustCompile(`.*unexpected state 'Failed', wanted target 'InService'.*`),
+			},
+			{
+				Config: testAccSagemakerEndpointConfigUpdate(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerEndpointExists("aws_sagemaker_endpoint.foo"),
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_endpoint.foo",
+						"name",
+						"terraform-testacc-sagemaker-endpoint-foo"),
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_endpoint.foo",
+						"endpoint_config_name",
+						fmt.Sprintf("terraform-testacc-sagemaker-endpoint-updated-%s", rName)),
+				),
+			},
+			{
+				ResourceName:      "aws_sagemaker_endpoint.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckSagemakerEndpointDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).sagemakerconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_sagemaker_endpoint" {
+			continue
+		}
+
+		resp, err := conn.ListEndpoints(&sagemaker.ListEndpointsInput{
+			NameContains: aws.String(rs.Primary.ID),
```

Comment:
Similar note as `aws_sagemaker_endpoint_configuration` resource, should prefer `conn.DescribeEndpoint()` here to remove potential false positives.